x/valset-pref/msg_server.go

```diff
@@ -49,6 +49,30 @@ func (server msgServer) DelegateToValidatorSet(goCtx context.Context, msg *types
 
 // UndelegateFromValidatorSet undelegates {coin} amount from the validator set.
 func (server msgServer) UndelegateFromValidatorSet(goCtx context.Context, msg *types.MsgUndelegateFromValidatorSet) (*types.MsgUndelegateFromValidatorSetResponse, error) {
+	// ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// err := server.keeper.UndelegateFromValidatorSet(ctx, msg.Delegator, msg.Coin)
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	return &types.MsgUndelegateFromValidatorSetResponse{}, fmt.Errorf("not implemented, utilize UndelegateFromRebalancedValidatorSet instead")
+}
+
+// UndelegateFromRebalancedValidatorSet undelegates {coin} amount from the validator set, utilizing a user's current delegations
+// to their validator set to determine the weights.
+func (server msgServer) UndelegateFromRebalancedValidatorSet(goCtx context.Context, msg *types.MsgUndelegateFromRebalancedValidatorSet) (*types.MsgUndelegateFromRebalancedValidatorSetResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	err := server.keeper.UndelegateFromRebalancedValidatorSet(ctx, msg.Delegator, msg.Coin)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.MsgUndelegateFromRebalancedValidatorSetResponse{}, nil
+}
+
+func (server msgServer) UndelegateProportionallyFromValidatorSet(goCtx context.Context, msg *types.MsgUndelegateFromValidatorSet) (*types.MsgUndelegateFromValidatorSetResponse, error) {
```

Comment:
Doesn't exposing this message mean that the disabled paths are accessible? My understanding is we want to delete this one and keep the commented one above unless I'm missing something

Reply:
No, we utilize a different method for this msgServer message. This method is the rebalanced undelegation. We need both msgs because they will do different things.

Reply:
Oops it was code duplication

Reply:
Removing it now, was accidental